### PR TITLE
feat: Support entire 2xx HTTP status range for webhooks

### LINKED ISSUE #14: Feature Request: Allow any 2xx status for webhooks
## Motivation
Currently, when a webhook does not respond with a `HTTP 200/OK` but instead e.g. a ´HTTP 204/No Content` the webhook is not considered successful.

This becomes an issue when e.g., trying to send to a discord webhook as the returned HTTP status is 204.

## Desired behavior

Any 2xx status code is considered successful

### PR DESCRIPTION
**Pull request by timo-reymann**
Fixes #14

### CODE REVIEW

File: webhook.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func (wh *Webhook) Send(ctx context.Context, destination, text string) error {
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
```
Comment (paskal) [inline webhook.go]: Good riddance! Can you please convert it to "less than `http.StatusOK`" and "bigger than `http.StatusMultipleChoices`", and write tests for `http.Status.Ok -1`, `http.StatusMultipleChoices -1`, `http.StatusMultipleChoices` and `http.StatusMultipleChoices + 1`?

**Review by paskal — CHANGES_REQUESTED** [review]

File: webhook.go (hunk 1 above)
Comment (timo-reymann) [reply webhook.go]: Done, also grouped tests into `t.Run` calls and added helper methods for the simple "status => error, no error"

Comment (timo-reymann) [reply webhook.go]: image

**Review by paskal — APPROVED** [review]
Thank you!

**Review by paskal — APPROVED** [review]
@umputun please squash on merge instead of rebasing.

**Comment by paskal** [comment]
@timo-reymann your change is released as v1.2.0

**Comment by timo-reymann** [comment]
Thank you! 🚀

**Pull request merged**